src/dataflow/src/source/kafka.rs

```diff
@@ -390,7 +403,31 @@ impl KafkaSourceReader {
                     old: self.last_stats.take(),
                     new: Some(stats.clone()),
                 });
-                self.last_stats = Some(stats);
+                self.last_stats = Some(stats.clone());
+            }
+            let mut cursor = std::io::Cursor::new(vec![]);
+            let write_res = stats.as_ref().to_writer(&mut cursor);
+            // If the write failed log an error and continue to the next loop iteration
+            if let Err(e) = write_res {
+                error!("error serializing rdkafka stats: {}", e);
+                continue;
+            }
+
+            // This temp var is required to satisfy the type checker
+            let res: Result<Statistics, serde_json::Error> = serde_json::from_reader(cursor);
```

Comment:
can we deserialize in the callback instead? Seems like we unconditionally try to here

that also, as far as I can tell, will move the possibly-expensive work on deserialization onto the kafka threads, and off the timely threads, which seems advantageous?

Reply:
We can assuming its valid to set a callback of each type since the existing codepath here expects to get the raw json. I agree though that it would be best done in rdkafka.

Reply:
With the way everything is setup right now this would be a more intrusive effort than I think its worth considering we also need the raw json. I refactored this version to not use the circuitous route through JsonbRef and std::io::Cursor but instead just deserialize from the jsonb more directly.

I would like to do a larger refactor to move the stats processing out of the dataflow and into a tokio task but that will require more time.

Reply:
The default statistics callback interval is 1s, so it is definitely not going to show up on profiles. Leaving it here seems 100% fine.